### ISSUE TITLE
Memory leak / high CPU

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#1036]
Repro:

``` clojure
$ clj -J-Xmx1g
Clojure 1.10.1
user=> (require '[clj-kondo.core :as clj-kondo])
nil
user=> (dotimes [i 10] (do (clj-kondo/run! {:lint ["src"]})))
nil
user=> (dotimes [i 10] (do (clj-kondo/run! {:lint ["src"]})))
nil
user=> (dotimes [i 10] (do (clj-kondo/run! {:lint ["src"]})))
nil
user=> (dotimes [i 10] (do (clj-kondo/run! {:lint ["src"]})))
nil
user=> (dotimes [i 10] (do (clj-kondo/run! {:lint ["src"]})))
user=> (dotimes [i 10] (do (clj-kondo/run! {:lint ["src"]})))
Execution error (OutOfMemoryError) at (REPL:1).
Java heap space
``` 

This is reported by VSCode users where clj-kondo is running as a long lived process.

Cause: we were holding on to a `ctx` argument in the hooks namespace by memoizing over it, which contains the entire linting state.

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/edfe1ff07cb99ae6370a5f7760aa757cd9ca95bc.

Proof:

``` 
$ clj -J-Xmx1g
Clojure 1.10.1
user=> (require '[clj-kondo.core :as clj-kondo])
nil
user=> (do (dotimes [i 100] (prn i) (clj-kondo/run! {:lint ["src"]})) (System/gc))
``` 

Monitor the Java process:

``` 
$ jps -m
$ top -pid 43477
``` 

No OOM, memory is back to reasonable amount after final GC.
```

**Comment by borkdude** [comment]
Update to `2020.09.10-20201007.185225-22` to fix the problem.
